src/components/HomePageModal/index.tsx

```diff
@@ -0,0 +1,141 @@
+import React, {
+  useCallback,
+  useEffect,
+  useState,
+  FormEvent,
+  useMemo,
+} from 'react';
+import { SingleValue } from 'react-select';
+import Selector, { SelectOption } from 'components/Selector';
+import { countries } from './listOfCountries';
+import LanguageSelect from 'components/LanguageSelect';
+import Copyright from 'components/copyright';
+import { Language } from 'types';
+import styles from './index.module.scss';
+
+const HomePageModal = (): JSX.Element => {
+  const [selectedCountry, setSelectedCountry] = useState<
+    SelectOption | undefined
+  >();
+  const [selectedLanguage, setSelectedLanguage] = useState<Language>(
+    Language.English,
+  );
+  const [showAlert, setShowAlert] = useState(false);
+  const [showModal, setShowModal] = useState(true);
+
+  useEffect(() => {
+    try {
+      const serializedState = localStorage.getItem('isShowModalWindow');
+      if (serializedState) {
+        setShowModal(false);
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  }, []);
+
+  const selectorWrapperClassName = useMemo(
+    () => (showAlert ? styles.selectorWrapperAlert : styles.selectorWrapper),
+    [showAlert],
+  );
+
+  const handleCountryChange = useCallback(
+    (newValue: SingleValue<SelectOption | undefined>) => {
+      const country = newValue as SelectOption;
+      setShowAlert(false);
+      setSelectedCountry(country);
+    },
+    [setShowAlert, setSelectedCountry],
+  );
+
+  const handleLanguageChange = useCallback(
+    (language: Language) => () => setSelectedLanguage(language),
+    [setSelectedLanguage],
+  );
+
+  const languageButtonClassName = useCallback(
+    (language: string) =>
+      language === selectedLanguage ? styles.focus : styles.languageButton,
+    [],
+  );
+
+  const saveCountryLanguage = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const checkboxChecked = (event.target as HTMLFormElement).check.checked;
+
+    if (!showModal) return null;
```

Comment:
It seems `if (!showModal) return null;` and the if-else block separately looks clear